src/core/tests/type_prop/reshape.cpp

```diff
@@ -20,6 +20,18 @@ TEST(type_prop, static_value_propagation) {
     ASSERT_EQ(r->get_shape(), (Shape{1, 2, 3}));
 }
 
+TEST(type_prop, tricky_label_propagation_for_auto_batch_case) {
+    auto shape = ov::PartialShape({1, 1280, 1, 1});
+    ov::DimensionTracker::set_label(shape[0], 1);
+    auto param = make_shared<op::Parameter>(element::f32, shape);
+    auto pattern = op::v0::Constant::create(element::i64, {2}, {-1, 1280});
+    auto r = make_shared<op::v1::Reshape>(param, pattern, false);
+
+    ASSERT_EQ(r->get_element_type(), element::f32);
+    ASSERT_EQ(r->get_shape(), (Shape{1, 1280}));
+    ASSERT_EQ(1, ov::DimensionTracker::get_label(r->get_output_partial_shape(0)[0]));
+}
+
```

Comment:
Consider somethin similar to:
```suggestion
TEST(type_prop, reshape_label_propagation_minus_one_special_zero_case_1) {
    auto data_shape = PartialShape{4, -1, 2, 1, 3};
    set_shape_labels(data_shape, 10);

    auto input = make_shared<op::v0::Parameter>(element::f32, data_shape);
    auto output_pattern = make_shared<op::v0::Constant>(element::i64, Shape{3}, std::vector<int64_t>{-1, 12, 0});
    output_pattern->get_default_output().get_tensor().set_value_label({20, 21, 22});

    const auto reshape = make_shared<op::v1::Reshape>(input, output_pattern, true);

    auto output_shape = reshape->get_output_partial_shape(0);
    EXPECT_EQ(output_shape, PartialShape({-1, 12, 2}));
    EXPECT_THAT(get_shape_labels(output_shape), ElementsAre(ov::no_label, 21, 12));
}
```

With previous version the labels on output were, losing label will be ok now?
```    EXPECT_THAT(get_shape_labels(output_shape), ElementsAre(11, 21, 12));```